File: app/channels/mentor_request_channel.rb
class MentorRequestChannel < ApplicationCable::Channel
  def subscribed
    mentor_request = Mentor::Request.find_by!(uuid: params[:uuid])

    stream_for mentor_request
  end

  def unsubscribed; end

  def self.broadcast!(request)
    broadcast_to request, {
      mentor_request: {
        uuid: request.uuid,
        status: request.status
      }
    }
  end
end

File: app/javascript/channels/mentorRequestChannel.ts
import consumer from '../utils/action-cable-consumer'
import { MentorSessionRequest } from '../components/types'

export type ChannelResponse = {
  uuid: string
  status: 'cancelled' | 'pending' | 'fulfilled'
}
export class MentorRequestChannel {
  subscription: ActionCable.Channel

  constructor(
    request: MentorSessionRequest,
    onReceive: (response: ChannelResponse) => void
  ) {
    this.subscription = consumer.subscriptions.create(
      {
        channel: 'MentorRequestChannel',
        uuid: request.uuid,
      },
      {
        received: (response: ChannelResponse) => {
          onReceive(response)
        },
      }
    )
  }

  disconnect(): void {
    this.subscription.unsubscribe()
  }
}
